src/linker/Linker.Dataflow/FlowAnnotations.cs

```diff
@@ -28,12 +28,16 @@ public FlowAnnotations (LinkContext context)
 		}
 
 		public bool RequiresDataFlowAnalysis (MethodDefinition method) =>
+			GetAnnotations (method.DeclaringType).TryGetAnnotation (method, out var methodAnnotations)
+				&& (methodAnnotations.ReturnParameterAnnotation != DynamicallyAccessedMemberTypes.None || methodAnnotations.ParameterAnnotations != null);
+
+		public bool HasAnnotations (MethodDefinition method) =>
 			GetAnnotations (method.DeclaringType).TryGetAnnotation (method, out _);
 
 		public bool RequiresDataFlowAnalysis (FieldDefinition field) =>
 			GetAnnotations (field.DeclaringType).TryGetAnnotation (field, out _);
 
-		public bool RequiresDataFlowAnalysis (GenericParameter genericParameter) =>
+		public bool HasAnnotations (GenericParameter genericParameter) =>
```

Comment:
Similarly - "has annotations" is not very telling in the callsite (it has a meaning in this class, but not externally). So maybe `RequiresGenericArgumentDataFlow` ?

Reply:
I was modeling it after the `GenericArgumentDataFlow`, but if we change that, then I would also change the name of the method respectively. So `RequiresAnalysis` works ... with `GenericArgumentAnalysis`? I guess that would work.
Personally I don't mind calling it `GenericArgumentDataFlow` even though it's not really data-flow. We're already using DataFlow as the term for everything "figure out reflection" related anyway.